sktime/forecasting/timemoe.py

```diff
@@ -0,0 +1,395 @@
+"""Implements TimeMOE forecaster."""
+
+__author__ = ["Maple728", "KimMeen", "PranavBhatP"]
+# Maple728 and KimMeen for timemoe
+__all__ = ["TimeMoEForecaster"]
+
+
+import numpy as np
+import pandas as pd
+from skbase.utils.dependencies import _check_soft_dependencies
+
+from sktime.forecasting.base import _BaseGlobalForecaster
+from sktime.utils.singleton import _multiton
+
+if _check_soft_dependencies("torch", severity="none"):
```

Comment:
the dependencies could all be moved inside the class - that is more robust against changes in the soft dependencies.

For the `config`, you could have a private method `_get_default_config` that returns the current dict.